### DISCUSSION TITLE
ESP32 - Wiznet W5200 problem

### DISCUSSION THREAD

**Discussion by victorsonntag** [khoih-prog/Ethernet_Generic#13] [General]
Hi @khoih-prog ,

Thanks for your great code!
I am working on a project (http webserber on an esp32 WROOM) and I use your great library. It is works perfectly with w5500 (wiz850io), but when I use with w5200 (wiz820io) the Ethernet.hardwareStatus() come back with 'EthernetNoHardware' value. The same setup and code works without any problem with 'ethernet.h' library.  I compared the two library and I found a difference in the init() void. In your code there is an extra 'softReset()' command, which is not in 'w5100.cpp'. After I commented out, the problem is solved. 
I don't know why it's causing problem on w5200, but I hope this information helps you to fix this bug.

Please let me know if you need any further information!

**Comment by khoih-prog** [comment]
Hi @victorsonntag 

Thanks for the **good** investigation and info.

I actually never have and use `W5200` as it's so unpopular and don't know where to buy. The library was then tested with `W5100`, `W5100S` and `W5500`.

Thanks to this fix, I'll make change in the new release to support `W5200`.

Regards,

**Reply by khoih-prog** [reply]
Hi @victorsonntag 

I'd appreciate if you can make a PR, I'm so busy now.

**Comment by khoih-prog** [comment]
HI @victorsonntag 

Are you sure you're using this [**Ethernet_Generic** library](https://github.com/khoih-prog/Ethernet_Generic) and the latest version **v2.6.0**? I can't find anywhere in the library the mentioned `init()` using `softreset()`

If you mean this

https://github.com/khoih-prog/Ethernet_Generic/blob/f8fa9336752d0796776ad1a4aadd8a3c9c65b2cc/examples/WebClientRepeating_ESP/WebClientRepeating_ESP.ino#L100

then you have to change it to use the correct `CS` pin, according to your wiring.

**Comment by khoih-prog** [comment]
Do you mean this line, in [**w5100_Impl.h**](https://github.com/khoih-prog/Ethernet_Generic/blob/main/src/utility/w5100_Impl.h)

https://github.com/khoih-prog/Ethernet_Generic/blob/f8fa9336752d0796776ad1a4aadd8a3c9c65b2cc/src/utility/w5100_Impl.h#L331

**Reply by victorsonntag** [reply]
Yes, this is the one

**Comment by khoih-prog** [comment]
Hi @victorsonntag 

The [**Ethernet_Generic** v2.6.1](https://github.com/khoih-prog/Ethernet_Generic/releases/tag/v2.6.1) has just been published.
Your contribution is noted in [**Contributions and Thanks**](https://github.com/khoih-prog/Ethernet_Generic#contributions-and-thanks)

Please test to be sure it's working with W5200

Regards,

---

### Releases v2.6.1

1. Fix bug for `Wiznet W5200 / wiz820io`. Check [ESP32 - Wiznet W5200 problem #13](https://github.com/khoih-prog/Ethernet_Generic/discussions/13)
2. Update `Packages' Patches`

**Reply by victorsonntag** [reply]
Hi @khoih-prog 

I tried it and it works perfectly with w5200 and w5500 too. Thank you for the quick update!
